server/src/test/java/io/crate/integrationtests/CircuitBreakerIntegrationTest.java

```diff
@@ -65,9 +71,10 @@ public void testQueryBreakerIsUpdatedWhenSettingIsChanged() {
         execute("select text from t1 group by text");
 
         execute("set global \"indices.breaker.query.limit\"='100b'");
-        expectedException.expect(SQLActionException.class);
-        expectedException.expectMessage("CircuitBreakingException: [query] Data too large, data for [collect: 0] " +
-                                        "would be [130/130b], which is larger than the limit of [100/100b]");
         execute("select text from t1 group by text");
+
+        var message = "[query] Data too large, data for [collect: 0] would be [130/130b], which is larger than the limit of [100/100b]";
```

Comment:
@mfussenegger Before i start migration 300+ tests i would get your opinion on this way of testing. What do you think ?

Reply:
How about something like `isSqlError(message, pgCode, httpCode)` ? 

Would force us to always check both. 


Btw. I think we may also consider to change the tests from using `expectedException` to `assertThrows`.

`ExcpectedException` is deprecated in junit 4.13: (We're still on 4.12) 

```
     * @deprecated Since 4.13
     * {@link org.junit.Assert#assertThrows(Class, org.junit.function.ThrowingRunnable)
     * Assert.assertThrows} can be used to verify that your code throws a specific
     * exception.
```

Reply:
`ExpectedException` would be gone anyway in all response related errors.

Reply:
May be a bit trappy to turn the error case into the normal result set. If we have statement where we don't verify the response at all we also wouldn't notice if the statement failed.

E.g. for refreshes, create table (where we have a syntax error or so on) that may make it more troublesome to write new tests.

Reply:
I see. Do you think i should go ahead with the current approach and start migrating all the tests or do you think this still needs some conceptional iterations ?

Reply:
The original exception is still part of the error classes, so i could be rethrown on request. Something like `response.throwException()`

Reply:
> The original exception is still part of the error classes, so it could be rethrown on request. Something like `response.throwException()`

Yep, but that requires an explicit action. I think it's safer to raise an exception by default. 

Could we have a `assertThrows(statement, isSqlError(message, pgCode, httpCode))` or something like that?

Reply:
I see what i can do, thanks for the feedback.